partner_customer_supplier/models/res_partner.py

```diff
@@ -0,0 +1,60 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+DOMAIN_SEARCH = {
+    ("=", False): ("=", 0),
+    ("=", True): (">=", 1),
+    ("!=", False): (">=", 1),
+    ("!=", True): ("=", 0),
+}
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    customer = fields.Boolean(
+        string="Is Customer",
+        compute="_compute_partner_is_customer",
+        inverse="_inverse_partner_is_customer",
+        search="_search_partner_is_customer",
+    )
+    supplier = fields.Boolean(
+        string="Is Vendor",
+        compute="_compute_partner_is_supplier",
+        inverse="_inverse_partner_is_supplier",
+        search="_search_partner_is_supplier",
+    )
+
+    @api.depends("customer_rank")
+    def _compute_partner_is_customer(self):
+        for partner in self:
+            partner.customer = partner.customer_rank > 0
```

Comment:
but this will be inconsistent, since the _increase_rank is done by sql, so that do not triggers the compute leading on an inconsistency between the value on DB and the value on your new field right?

Reply:
No, because this field is not stored, so its computed each time is consulted